src/sentry/sentry_apps/external_requests/issue_link_requester.py

```diff
@@ -74,26 +73,43 @@ def run(self) -> dict[str, Any]:
             body = safe_urlread(request)
             response = json.loads(body)
         except (json.JSONDecodeError, TypeError):
-            raise ValidationError(f"Unable to parse response from {self.sentry_app.slug}")
-        except Exception as e:
-            logger.info(
-                "issue-link-requester.error",
-                extra={
-                    "sentry_app": self.sentry_app.slug,
-                    "install": self.install.uuid,
-                    "project": self.group.project.slug,
-                    "group": self.group.id,
+            raise SentryAppIntegratorError(
+                message=f"Unable to parse response from {self.sentry_app.slug}",
+                webhook_context={
+                    "error_type": "issue-link-requester.invalid-json",
                     "uri": self.uri,
-                    "error_message": str(e),
+                    "response": body,
+                    "installation_uuid": self.install.uuid,
                 },
+                status_code=503,
             )
-            raise APIError(
-                f"Issue occured while trying to contact {self.sentry_app.slug} to link issue"
+        except Exception as e:
+            error_type = "issue-link-requester.error"
+            extras = {
+                "sentry_app": self.sentry_app.slug,
+                "installation_uuid": self.install.uuid,
+                "project": self.group.project.slug,
+                "group": self.group.id,
+                "uri": self.uri,
+                "error_message": str(e),
+            }
+            logger.info(error_type, extra=extras)
+            raise SentryAppIntegratorError(
+                message=f"Issue occured while trying to contact {self.sentry_app.slug} to link issue",
+                webhook_context={"error_type": error_type, **extras},
+                status_code=503,
```

Comment:
should this be 500? normal exceptions that get raised and are uncaught by django error handling return 500

Reply:
We can catch `RequestExceptions` here instead of all Exceptions. It's probably better to have the uncaughts go through so we can get alerts and fine tune. Alternatively, we could have other non RequestException exceptions go under `SentryAppSentryError` which would also return a 500 and be a bit nicer